clients/src/main/java/org/apache/kafka/common/utils/EventQueue.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.common.utils;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+
+public interface EventQueue extends AutoCloseable {
+    interface Event<T> {
+        T run() throws Throwable;
+    }
```

Comment:
> Can it throws Exception instead?

Good idea.

>  ...maybe it's just interface Event<T> extends Callable<T> {}

Does extending `Callable` help here?  I'm not really sure (I might be missing something).  Java lets you create lambdas with any one-function interface.  I was thinking we might add more methods later, which is why I didn't just use `Callable` in the first place...  I'll leave it alone for now.

Reply:
> it simplifies the signature of KafkaEventQueue.EventHandler.handleEvents() because it can throws InterruptedException instead of throws Throwable, and then EventContext.run() no longer throws anything.

good point.  done